File: cpp/128.longest-consecutive-sequence.cpp
/*
 * @lc app=leetcode id=128 lang=cpp
 *
 * [128] Longest Consecutive Sequence
 */

// @lc code=start
class Solution {
public:
    // int longestConsecutive(vector<int>& nums) {
    //     if(nums.size()==0)return 0;
    //     sort(begin(nums),end(nums));
    //     int res=1;
    //     int count=1;
    //     for(int i=0;i<nums.size()-1;i++){
    //         if(nums[i] == nums[i+1]) continue;
    //         if(nums[i+1]-nums[i]==1)
    //             count++;
    //         else {
    //             res=max(res,count);
    //             count=1;
    //         }
    //     }
    //     return max(res,count);
    // }

    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> data(begin(nums), end(nums));
        int res = 0;
        for(int n : nums){
            if(!data.count(n-1)){
                int count = 0;
                while(data.count(n++)) count++;
                res = max(res, count);
            }
        }
        return res;
    }
};
// @lc code=end

File: python3/128.longest-consecutive-sequence.py
#
# @lc app=leetcode id=128 lang=python3
#
# [128] Longest Consecutive Sequence
#

# @lc code=start
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        nums = set(nums)
        for num in nums:
            if num - 1 not in nums:
                next = num + 1
                while next in nums:
                    next += 1
                res = max(res, next-num)
        return res
# @lc code=end

